scripts/variantstore/wdl/GvsWithdrawSamples.wdl

```diff
@@ -77,22 +78,36 @@ task WithdrawSamples {
     fi
 
     # create the temp table (expires in 1 day)
-    bq --apilog=false --project_id=~{project_id} mk --expiration=86400 ~{dataset_name}.current_callset_samples "sample_name:STRING"
+    TEMP_TABLE_NAME="~{dataset_name}.current_call_set_samples_$(date +%s)"
+    echo "Creating Temp Table: $TEMP_TABLE_NAME"
+    bq --apilog=false --project_id=~{project_id} mk --expiration=86400 $TEMP_TABLE_NAME "sample_name:STRING"
+
     # populate the temp table
-    bq --apilog=false load --project_id=~{project_id} -F "tab" ~{dataset_name}.current_callset_samples sample_names.tsv
+    echo "Populating Temp Table: $TEMP_TABLE_NAME"
+    bq --apilog=false load --project_id=~{project_id} -F "tab" $TEMP_TABLE_NAME sample_names.tsv
 
-    # join on the temp table to figure out which samples should be marked as withdrawn
+    # Update sample_info.withdrawn by joining on the temp table to figure out which samples should be marked as withdrawn
+    echo "Updating samples that should be withdrawn"
     bq --apilog=false --project_id=~{project_id} query --format=csv --use_legacy_sql=false \
-      'UPDATE `~{dataset_name}.sample_info` AS samples SET withdrawn = "~{withdrawn_timestamp}" \
-        WHERE NOT EXISTS \
-        (SELECT * \
-        FROM `~{project_id}.~{dataset_name}.current_callset_samples` AS callset \
-        WHERE \
-        samples.sample_name = callset.sample_name) \
-        AND NOT samples.is_control \
+      'UPDATE `~{dataset_name}.sample_info` AS samples SET withdrawn = "~{withdrawn_timestamp}"
+        WHERE NOT EXISTS
+          (SELECT * FROM `~{project_id}.'"${TEMP_TABLE_NAME}"'` AS callset
+            WHERE samples.sample_name = callset.sample_name)
+        AND NOT samples.is_control
         AND withdrawn IS NULL' > log_message.txt
 
     cat log_message.txt | sed -e 's/Number of affected rows: //' > rows_updated.txt
+
+    # Now, determine if there are any samples in the uploaded list that are NOT in sample_info and report this
+    echo "Determining if there are any new samples that should be upladed"
+    bq --apilog=false --project_id=gvs-internal query --format=csv --use_legacy_sql=false \
+      'SELECT callset.sample_name
+        FROM `~{project_id}.'"${TEMP_TABLE_NAME}"'` callset
+        LEFT JOIN `~{dataset_name}.sample_info` sample_info ON sample_info.sample_name = callset.sample_name
+        WHERE sample_info.sample_name IS NULL' > new_samples.txt
+
+    echo "The following samples in the uploaded list have NOT been loaded into ~{dataset_name}"
```

Comment:
I worry this message might get missed...could this condition be checked first and generate an error if there are samples that need to be loaded?

Reply:
I wondered about this too, but the AC are to let it pass if there are samples that have not been loaded.